crates/net/network/tests/it/txgossip.rs

```diff
@@ -42,3 +43,85 @@ async fn test_tx_gossip() {
     let received = peer1_tx_listener.recv().await.unwrap();
     assert_eq!(received, hash);
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_4844_tx_gossip_penalization() {
+    reth_tracing::init_test_tracing();
+    let provider = MockEthProvider::default();
+    let net = Testnet::create_with(2, provider.clone()).await;
+
+    // install request handlers
+    let net = net.with_eth_pool();
+
+    let handle = net.spawn();
+
+    let peer0 = &handle.peers()[0];
+    let peer1 = &handle.peers()[1];
+
+    // connect all the peers
+    handle.connect_peers().await;
+
+    let mut peer1_tx_listener = peer1.pool().unwrap().pending_transactions_listener();
+
+    let mut gen = TransactionGenerator::new(thread_rng());
+
+    // peer 0 will be penalised for sending txs[0] over gossip
+    let txs = vec![gen.gen_eip4844_pooled(), gen.gen_eip1559_pooled()];
+
+    txs.iter().for_each(|tx| {
+        let sender = tx.sender();
+        provider.add_account(sender, ExtendedAccount::new(0, U256::from(100_000_000)));
+    });
+
+    let signed_txs: Vec<Arc<TransactionSigned>> =
+        txs.iter().map(|tx| Arc::new(tx.transaction().clone().into_signed())).collect();
+
+    let network_handle = peer0.network();
+
+    let peer0_reputation_before =
+        peer1.peer_handle().peer_by_id(peer0.peer_id().clone()).await.unwrap().reputation();
+
+    network_handle.send_transactions(peer1.peer_id().clone(), signed_txs);
+
+    tokio::time::sleep(Duration::from_secs(1)).await;
+
+    let received = peer1_tx_listener.recv().await.unwrap();
```

Comment:
```suggestion

    let received = peer1_tx_listener.recv().await.unwrap();
```

here the sleep isn't necessary since the `await` syntax is used